streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -428,7 +469,7 @@ private void rebuildMetadata(final Map<HostInfo, Set<TopicPartition>> activePart
         final Set<HostInfo> standbyHosts = new HashSet<>();
         for (final StreamsMetadata streamsMetadata : allMetadata) {
             if (streamsMetadata instanceof NamedTopologyStreamsMetadataImpl
-                && ((NamedTopologyStreamsMetadataImpl) streamsMetadata).namedTopologyToStoreNames().get(topologyName).contains(storeName)) {
+                && ((NamedTopologyStreamsMetadataImpl) streamsMetadata).storeNamesByTopology().get(topologyName).contains(storeName)) {
```

Comment:
Hmm.. this seems not right to me. Below we want to collect both active and standby host infos in the `KeyQueryMetadata`, but here we would only enter if the active store names contains it.

If that should be fixed, then it seems that `storeNamesByTopology()` and `standbyStoreNamesByTopology()` always appears hand-in-hand, hence I'm wondering if it's necessary to keep them in two fields and two getter functions?

Reply:
This comment is the lines 473 - 476 below: should we use the `topologyName` as well? Otherwise, e.g. 

```
final Set<String> standbyStateStoreNames = streamsMetadata.standbyStateStoreNames();
```

Would be getting for the `UNNAMED` topology, which seems not correct?

Reply:
See the latest commit, I refactored how we handle this so that each StreamsMetadata object is associated with a single `topologyName`